src/commands/build.ts

```diff
@@ -285,99 +277,69 @@ export async function command(commandOptions: CommandOptions) {
-                webModulesLoc,
-                dependencyImportMap.imports[spec],
-              );
-              const extName = path.extname(resolvedImport);
-              if (!isBundled && extName && extName !== '.js') {
-                resolvedImport = resolvedImport + '.proxy.js';
               }
-              return resolvedImport;
+              return resolvedImportUrl;
             }
+            // If that fails, return a placeholder import and attempt to resolve.
             let [missingPackageName, ...deepPackagePathParts] = spec.split('/');
             if (missingPackageName.startsWith('@')) {
               missingPackageName += '/' + deepPackagePathParts.shift();
             }
             messageBus.emit('MISSING_WEB_MODULE', {
-              id: f,
+              id: fileLoc,
               data: {
                 spec: spec,
                 pkgName: missingPackageName,
               },
             });
-            return path.posix.join(config.buildOptions.baseUrl, webModulesLoc, `${spec}.js`);
+            // Sort of lazy, but we expect "MISSING_WEB_MODULE" to exit the build with an error.
+            // So, just return the original import here since it will never be seen.
+            return spec;
           });
           code = wrapImportMeta({code, env: true, hmr: false, config});
         }
         await fs.mkdir(path.dirname(outPath), {recursive: true});
         await fs.writeFile(outPath, code);
-        allBuiltFromFiles.add(f);
+        allBuiltFromFiles.add(fileLoc);
       }
     }
     messageBus.emit('WORKER_COMPLETE', {id, error: null});
   }
 
   stopEsbuild();
-
-  for (const proxiedFileLoc of allCssModules) {
-    const proxiedCode = await fs.readFile(proxiedFileLoc, {encoding: 'utf8'});
-    const proxiedExt = path.extname(proxiedFileLoc);
-    const proxiedUrl = proxiedFileLoc.substr(buildDirectoryLoc.length);
-    const proxyCode = await wrapCssModuleResponse({
-      url: proxiedUrl,
-      code: proxiedCode,
-      ext: proxiedExt,
-      config,
-    });
-    const proxyFileLoc = proxiedFileLoc.replace('.module.css', '.css.module.js');
-    await fs.writeFile(proxyFileLoc, proxyCode, {encoding: 'utf8'});
-  }
   for (const proxiedFileLoc of allProxiedFiles) {
     const proxiedCode = await fs.readFile(proxiedFileLoc, {encoding: 'utf8'});
     const proxiedExt = path.extname(proxiedFileLoc);
     const proxiedUrl = proxiedFileLoc.substr(buildDirectoryLoc.length);
-    const proxyCode = wrapEsmProxyResponse({
-      url: proxiedUrl,
-      code: proxiedCode,
-      ext: proxiedExt,
-      config,
-    });
+    const proxyCode = proxiedFileLoc.endsWith('.module.css')
+      ? await wrapCssModuleResponse({
+          url: proxiedUrl,
+          code: proxiedCode,
+          ext: proxiedExt,
+          config,
+        })
+      : wrapEsmProxyResponse({
+          url: proxiedUrl,
+          code: proxiedCode,
+          ext: proxiedExt,
+          config,
+        });
```

Comment:
Trying to follow along here, but doesn’t this turn files that were once `*.css.module.js` into `*.module.css.proxy.js`? (looking at the `for… of allCssModules` block removed)

Reply:
yea, it's a bit verbose but better than adding custom logic to handle this one file type.